pgmpy/factors/CPD.py

```diff
@@ -116,15 +119,33 @@ def __init__(self, variable, variable_card, values,
 
         super(TabularCPD, self).__init__(variables, cardinality, values.flatten('C'))
 
-    def get_cpd(self):
+    def get_cpd(self, pretty=True):
         """
         Returns the cpd
         """
         if self.variable in self.variables:
+            if pretty:
+                table_list = []
+                for i in range(len(self.cardinality))[1:]:
+                    var_rep = np.prod(self.evidence_card[i:])
+                    row = [self.scope()[-i]]
+                    row.extend(self.variables[row[0]] * var_rep)
+                    if i > 1:
+                        row[1:] = [x for item in row[1:] for x in
+                                   repeat(item, int(np.prod(self.evidence_card[0:i - 1])))]
```

Comment:
@kislayabhi: Maybe try something in these lines:
```python
indexes = np.array(list(product(*[range(i) for i in self.cardinality[1:]])))
```
and then have a loop on columns of `indexes` to replace them:
```python
a[:, 0] = np.array(self.variables['iq'])[np.atleast_2d(a[:, 0])]
```
And then finally get the transpose and then could use `np.column_stack` to append the variable names.

I feel that this method should be a cleaner. What do you think?